Mjbrown

markjbrown opened the pull request:
## Description

Fixed a number of bugs in these scripts. 
Cleaned up variables and other changes intended to make the scripts easier to understand and use.
Added unique string generator for all resources requiring unique names.

## Checklist

<!--
    Filling in this checklist is mandatory! If you don't, your pull request
    will be rejected without further review. Checklists must be completed
    within 7 days of PR submission.

    To check a box in markdown, make sure that it is formatted as [X] (no whitespace).
    Not formatting checkboxes correctly may break automated tools and delay PR processing.
-->

- [X] Scripts in this pull request are written for the `bash` shell.
- [X] This pull request was tested on __at least one of__ the following platforms:
  - [ ] Linux
  - [X] Azure Cloud Shell
  - [ ] macOS
  - [X] Windows Subsystem for Linux
- [X] Scripts do not contain passwords or other secret tokens.
- [X] No deprecated commands or arguments are used. ([Release notes](https://docs.microsoft.com/cli/azure/release-notes-azure-cli))
- [X] All Azure resource identifiers which must be universally unique are guaranteed to be so.

### Testing information

CLI version: 2.0.67
```
az --version
```

Extensions required:

markjbrown commented:
#sign-off

sptramer commented:
@markjbrown Can merge after conflicts are resolved.

sptramer commented:
@markjbrown PR still indicates that there are conflicts with master - please resolve, and can be merged.

markjbrown commented:
Ack, merged the conflict. Good to go now! @sptramer Feel free to merge when ready.

Thanks.

sptramer commented on cosmosdb/secure-cosmosdb-get-connection-string/secure-cosmosdb-get-connection-string:
```diff
@@ -1,10 +1,12 @@
 #!/bin/bash
```
GitHub won't let me comment on the correct line, but generated the following errors:

```
az cosmosdb show: error: argument --resource-group/-g: expected one argument
usage: az cosmosdb show [-h] [--verbose] [--debug]
                        [--output {json,jsonc,table,tsv,yaml,none}]
                        [--query JMESPATH] [--subscription _SUBSCRIPTION]
                        [--resource-group RESOURCE_GROUP_NAME]
                        [--name ACCOUNT_NAME] [--ids ID [ID ...]]                                                                                             az cosmosdb list-keys: error: argument --resource-group/-g: expected one argument
usage: az cosmosdb list-keys [-h] [--verbose] [--debug]
                             [--output {json,jsonc,table,tsv,yaml,none}]
                             [--query JMESPATH] [--subscription _SUBSCRIPTION]
                             [--resource-group RESOURCE_GROUP_NAME]
                             [--name ACCOUNT_NAME] [--ids ID [ID ...]]
```

Looks like the script still uses `$myRG` in places when the variable is `$resourceGroupName`.

markjbrown replied:
fixed in next commit.

sptramer commented:
Tagging with `requires-rework` now that this repo has the correct labels on it.

markjbrown commented on cosmosdb/create-cosmosdb-gremlin-account/create-cosmosdb-gremlin-account.sh:
```diff
@@ -32,11 +37,11 @@ az cosmosdb database create \
-    --throughput 1000
\ No newline at end of file
+    --partition-key-path $partitionKeyPath \
+    --throughput throughput
```
These are removed.

markjbrown commented on cosmosdb/create-cosmosdb-mongodb-account/create-cosmosdb-mongodb-account.sh:
```diff
@@ -1,10 +1,16 @@
 #!/bin/bash
```
Fixed in next commit.

markjbrown commented on cosmosdb/create-cosmosdb-table-account/create-cosmosdb-table-account.sh:
```diff
@@ -1,20 +1,24 @@
 #!/bin/bash
```
Fixed in next commit.

markjbrown commented on cosmosdb/high-availability-cosmosdb-configure-failover/high-availability-cosmosdb-configure-failover.sh:
```diff
@@ -26,4 +29,4 @@ az cosmosdb create \
 	--resource-group $resourceGroupName \
-	--locations "South Central US"=0 "North Central US"=1 "East US"=2 "West US"=3
\ No newline at end of file
+	--locations "South Central US"=0 "North Central US"=1 "East US 2"=2 "West US 2"=3
```
Fixed anyway.

markjbrown commented:
Have addressed all the feedback above. Please review and merge ASAP. I have another large update to our CLI docs. Thanks.

markjbrown commented:
#sign-off

markjbrown commented on cosmosdb/create-cosmosdb-account-database/create-cosmosdb-account-database.sh:
```diff
@@ -1,20 +1,25 @@
+resourceGroupName='myResourceGroup-'$uniqueId
 location='southcentralus'
-accountName='myaccountname' #needs to be lower case
+accountName='mycosmosaccount-'$uniqueId #needs to be lower case
```
Fixed with shorter random string.

markjbrown commented on cosmosdb/create-cosmosdb-account-database/create-cosmosdb-account-database.sh:
```diff
@@ -1,20 +1,25 @@
+
 # Set variables for the new SQL API account, database, and container
-resourceGroupName='myResourceGroup'
+resourceGroupName='myResourceGroup-'$uniqueId
```
fixed.

markjbrown commented on cosmosdb/scale-cosmosdb-throughput/scale-cosmosdb-throughput.sh:
```diff
@@ -1,13 +1,17 @@
 #!/bin/bash
```
fixed.

sptramer commented on cosmosdb/scale-cosmosdb-replicate-multiple-regions/scale-cosmosdb-replicate-multiple-regions.sh:
```diff
@@ -1,10 +1,12 @@
 #!/bin/bash
```
__RECOMMENDED__: The script generates the following warning:

```
The regionName=failoverPriority method of specifying locations is deprecated. Use --locations KEY=VALUE [KEY=VALUE ...]
to specify the regionName, failoverPriority, and isZoneRedundant properties of the location.
Multiple locations can be specified by including more than one --locations argument.
```

This should be fixed in order to avoid the script breaking in a future release of the CLI.

markjbrown replied:
Updated in upcoming commit.

sptramer commented on cosmosdb/create-cosmosdb-gremlin-account/create-cosmosdb-gremlin-account.sh:
```diff
@@ -32,11 +37,11 @@ az cosmosdb database create \
-    --throughput 1000
\ No newline at end of file
+    --partition-key-path $partitionKeyPath \
+    --throughput throughput
```
Causes an error - should be `$throughput`?

markjbrown replied:
Thanks for catching. Fixed in next commit.

markjbrown commented:
I have made final edits/fixes for these scripts. Addressed all the non-linux friendly line endings. Also changed the failover script to correctly do a manual failover.

this should be good to commit.

thanks.

markjbrown commented:
#sign-off
